cql3/values.hh

```diff
@@ -86,38 +96,67 @@ public:
         return std::holds_alternative<unset_value>(_data);
     }
     bool is_value() const {
-        return std::holds_alternative<fragmented_temporary_buffer::view>(_data);
-    }
-    std::optional<fragmented_temporary_buffer::view> data() const {
-        if (auto pdata = std::get_if<fragmented_temporary_buffer::view>(&_data)) {
-            return *pdata;
-        }
-        return {};
+        return _data.index() <= 1;
     }
     explicit operator bool() const {
         return is_value();
     }
-    const fragmented_temporary_buffer::view* operator->() const {
-        return &std::get<fragmented_temporary_buffer::view>(_data);
+
+    template <typename Func>
+    requires std::invocable<Func, const managed_bytes_view&> && std::invocable<Func, const fragmented_temporary_buffer::view&>
+    decltype(auto) with_value(Func f) const {
+        switch (_data.index()) {
+        case 0: return f(std::get<fragmented_temporary_buffer::view>(_data));
+        default: return f(std::get<managed_bytes_view>(_data));
```

Comment:
Probably slightly faster to use std::get<const managed_bytes_view&>() to avoid copying the view (cheap as it is).

Reply:
std::get returns a reference, not a copy.